### PR TITLE
【CI】run all demo ci before exit in windows

### PR DESCRIPTION
**Pull request by betterpig**
### PR types

Others
### PR changes

Others
### Describe

- run all demo ci before exit in windows to get to know which test fails.
- package inference library even when inference library test failed.

### CODE REVIEW

File: paddle/fluid/inference/api/demo_ci/run.sh [hunk 1]
```diff
@@ -128,8 +128,8 @@ for WITH_STATIC_LIB in ON OFF; do
         --dirname=$DATA_DIR/word2vec/word2vec.inference.model \
         --use_gpu=$use_gpu
       if [ $? -ne 0 ]; then
-        echo "simple_on_word2vec demo runs fail."
-        exit 1
+        echo "simple_on_word2vec use_gpu:${use_gpu} runs failed " >> ${current_dir}/test_summary.txt
+        EXIT_CODE=1
```
Comment (zhwesky2010) [inline paddle/fluid/inference/api/demo_ci/run.sh:135]: EXIT_CODE在前面设默认值为0

File: paddle/fluid/inference/api/demo_ci/run.sh [hunk 2]
```diff
@@ -128,8 +128,8 @@ for WITH_STATIC_LIB in ON OFF; do
         --dirname=$DATA_DIR/word2vec/word2vec.inference.model \
         --use_gpu=$use_gpu
       if [ $? -ne 0 ]; then
-        echo "simple_on_word2vec demo runs fail."
-        exit 1
+        echo "simple_on_word2vec use_gpu:${use_gpu} runs failed " >> ${current_dir}/test_summary.txt
```
Comment (zhwesky2010) [inline paddle/fluid/inference/api/demo_ci/run.sh]: 第一个重定向 就 写 > 覆盖吧，或者先清理再创建一个test_summary.txt

File: paddle/scripts/paddle_build.bat [hunk 3]
```diff
@@ -267,8 +267,12 @@ call :cmake || goto cmake_error
 call :build || goto build_error
 call :test_whl_pacakage || goto test_whl_pacakage_error
 call :test_unit || goto test_unit_error
-::call :test_inference || goto test_inference_error
-::call :test_inference_ut || goto test_inference_ut_error
+::call :test_inference
```
Comment (zhwesky2010) [inline paddle/scripts/paddle_build.bat]: CI上还是挂了就退出，节省CI时间。这里不用改

File: paddle/scripts/paddle_build.bat [hunk 4]
```diff
@@ -760,12 +768,15 @@ for /F %%i in ("%libsize%") do (
...
 %cache_dir%\tools\busybox64.exe bash run.sh %work_dir:\=/% %WITH_MKL% %WITH_GPU% %cache_dir:\=/%/inference_demo %WITH_TENSORRT% %TENSORRT_ROOT% %WITH_ONNXRUNTIME% %MSVC_STATIC_CRT% "%CUDA_TOOLKIT_ROOT_DIR%"
+
 goto:eof
 
 :test_inference_error
 ::echo 1 > %cache_dir%\error_code.txt
 ::type %cache_dir%\error_code.txt
-echo Testing fluid library for inference failed!
+echo    ==========================================
+echo    Testing fluid library for inference failed!
```
Comment (zhwesky2010) [inline paddle/scripts/paddle_build.bat]: Testing inference library failed!

File: paddle/fluid/inference/api/demo_ci/run.sh (hunk 1 above)
Comment (betterpig) [reply paddle/fluid/inference/api/demo_ci/run.sh:135]: done

File: paddle/fluid/inference/api/demo_ci/run.sh (hunk 2 above)
Comment (betterpig) [reply paddle/fluid/inference/api/demo_ci/run.sh]: done

File: paddle/scripts/paddle_build.bat (hunk 4 above)
Comment (betterpig) [reply paddle/scripts/paddle_build.bat]: done

File: paddle/scripts/paddle_build.bat (hunk 3 above)
Comment (betterpig) [reply paddle/scripts/paddle_build.bat]: done

**Review by zhwesky2010 — APPROVED** [review]

**Comment by zhwesky2010** [comment]
LGTM

**Pull request merged**